### DISCUSSION TITLE
What's the input and the output of nni?

### DISCUSSION THREAD

**Discussion by xuehui1991** [microsoft/nni#3776] [Q&A] [answered]
**What would you like to be added**:
For now,  some user may confuse what's the input and the output of nni.
Such as NAS, shall we format one output format(output model? config? ) so that user won't confuse.

**Why is this needed**:

**Without this feature, how does current nni work**：

**Components that may involve changes**:

**Brief description of your proposal if any**:

**Comment by ultmaster** [comment, marked as answer]
Let's discuss it case by case.

For NAS, the derivation of final architecture is what is needed. For most algorithms, this post-processing might not be easy. So yes. I agree that an API should be designed to obtain the final model for user.
